### PR TITLE
avoid hard coded guest OS IDs

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR tries to avoid problems with guest os hypervisor mapping that we see in some DBs. If site specific entries were added the mapping may break on this.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [x] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
makes sense

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: engine/schema/src/main/java/com/cloud/upgrade/GuestOsMapper.java [hunk 1]
```diff
@@ -102,8 +102,16 @@ private boolean addGuestOs(long categoryId, String displayName) {
         guestOS = guestOSDao.persist(guestOS);
         return (guestOS != null);
     }
+    public void addGuestOsHypervisorMapping(GuestOSHypervisorMapping mapping, long category, String displayName) {
+        long guestOsId =  getGuestOsId(category, displayName);
+        if (guestOsId == 0) {
+            LOG.warn(String.format("no guest os found for category %d and name %s, skipping mapping it to %s/%s", guestOsId, displayName, mapping.getHypervisorType(), mapping.getHypervisorVersion()));
```
Comment (DaanHoogland) [inline engine/schema/src/main/java/com/cloud/upgrade/GuestOsMapper.java:108]:
```suggestion
            LOG.error(String.format("no guest os found for category %d and name %s, skipping mapping it to %s/%s", guestOsId, displayName, mapping.getHypervisorType(), mapping.getHypervisorVersion()));
```

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Review by borisstoyanov — APPROVED** [review]
LGTM, manually tested the changes

**Review by weizhouapache — APPROVED** [review]
code lgtm

makes perfect sense to remove hard-coded guest os id.

The last time we use hard-coded guest os id is in the upgrade to 4.13.0. If an environment was created after 4.13.0, it will be upgraded to latest 4.18 successfully, no mater if there are custom guest oses or mappings.

**Pull request merged**